crypto/evp/evp_pkey.c

```diff
@@ -73,6 +74,12 @@ EVP_PKEY *EVP_PKCS82PKEY_ex(const PKCS8_PRIV_KEY_INFO *p8, OSSL_LIB_CTX *libctx,
     int selection;
     size_t len;
     OSSL_DECODER_CTX *dctx = NULL;
+    const ASN1_OBJECT *algoid = NULL;
+    char keytype[OSSL_MAX_NAME_SIZE];
+
+    if (!PKCS8_pkey_get0(&algoid, NULL, NULL, NULL, p8)
```

Comment:
I wish we checked p8 == NULL here.. (Maybe off topic).

Reply:
Actually NOT off topic at all! I just checked. Although master uses the `p8` param without a NULL check, the `i2d_PKCS8_PRIV_KEY_INFO` call *does* check it (eventually....deep in the depths of the ASN1 code) and fails with a 0 return - causing this function to bail out. However, `PKCS8_pkey_get0` will seg fault if passed a NULL `p8` value. So we actually need an explicit check to keep the behaviour the same.